x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,212 @@
+  - **startDate**
+  - **endDate**
+  - **timeRangeKind**
+- **eqlOptions**
+- **indexNames**: List of indices in Timeline.
+- **id**: Timeline Id
+- **fields**:  List of requested fields
+- **filterQuery**:  combined query consisting: 
+  * **KQL/Lucene Query**
+  * **Filters**
+  * **Data Providers**
+- **runtimeMappings**: List of runtime mappings that need to be fetched
+- **language**: KQL | Lucene
+- **limit**: Sample Size or Batch Size
+- **sort**
+
+## Batch fetch/re-fetch Strategy
+Batch fetch/re-fetch strategy depends on the scenario or the action the user is taking. This section will lay out all the scenarios in which a batch should be re-fetched and its corresponding strategy.
+
+### New Batch Requested
+When a user requests a new batch after clicking on `Load more` as shown below, a new batch will be fetched. For example, if user is on Batch 0 and clicks on Load more, a new batch will be fetched with below strategy :
+
+![](./images/load_more_timeline.png)
+
+#### **Expected fetch/re-fetch logic**
+
+```js
+{
+  /* Fetch Batch 1 */
+  activePage: 1, /* current batch */
+  querySize: 500 /* batch/sample size */
+}
+```
+
+Consequently, if a user requests a new batch again, Batch 2 will be fetched with below strategy.
+
+```js
+{
+	/* Fetch Batch 2 */
+	activePage: 2,
+	querySize: 500 /* batch/sample size */
+}
+```
+#### **Expected UI behavior**
+
+- When a new batch is fetched, the Datagrid should have new pages appended to it.
+ 
+### Refresh Button
+
+When a user clicks on the Refresh button in the query tab, it should explicitly re-fetch 0th batch again irrespective of the fact if any batch parameters have been changed or not.
+
+This also applied to the refetch logic triggered by :
+- operations such as change in the Alert's status.
+- triggering of global query.
+- Clicking Refresh button in the query tab.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+#### Considerations
+- This logic might lead to degrdations in UX in scenario given below:
+  - when user is on let's say 3rd batch and they change the status of an Alert (which results in refetch). 
+  - This will reset the table page to go back to 1st since we re-fetch 0th batch as mentioned above. 
+  - Team decided that it is rare scenario since many users tend to do these kind of operations on the first page of the Datagrid.
+
+### Adding a new column
+Adding a new column should not affect the result set that is currently being displayed in front of the user. We should fetch a complete result set irrespective of the batch user is on.
+
+#### When user is on the first(0th) batch
+When user is on the first batch i.e. Batch 0, it will be re-fetched with the below strategy.
+
+- **Expected fetch/re-fetch logic**
+
+  ```js
+  {
+    /* Fetch Batch 0 */
+    activePage: 0,
+    querySize: 500 /* batch/sample size */
+  }
+  ```
+- **Expected UI behavior**
+
+  - User should see the new column added in the Datagrid.
```

Comment:
I think we need to be more specific here: are we staying on the same page or going back to page 1?

Reply:
I have reworded it. There will be no changes in the pagination as mentioned in: ba22b4e